packages/flutter_tools/lib/src/macos/xcode.dart

```diff
@@ -185,3 +189,303 @@ class Xcode {
     );
   }
 }
+
+/// A utility class for interacting with Xcode xcdevice command line tools.
+class XCDevice {
+  XCDevice({
+    @required ProcessManager processManager,
+    @required Logger logger,
+    @required Xcode xcode,
+  }) : _processUtils = ProcessUtils(logger: logger, processManager: processManager),
+       _logger = logger,
+       _xcode = xcode;
+
+  final ProcessUtils _processUtils;
+  final Logger _logger;
+  final Xcode _xcode;
+
+  bool get isInstalled => _xcode.isInstalledAndMeetsVersionCheck && xcdevicePath != null;
+
+  String _xcdevicePath;
+  String get xcdevicePath {
+    if (_xcdevicePath == null) {
+      try {
+        _xcdevicePath = _processUtils.runSync(
+          <String>[
+            'xcrun',
+            '--find',
+            'xcdevice'
+          ],
+        ).stdout.trim();
+      } on ProcessException {
+        // Ignored, return null below.
+      } on ArgumentError {
+        // Ignored, return null below.
+      }
+    }
+    return _xcdevicePath;
+  }
+
+  Future<List<dynamic>> _getAllDevices() async {
+    if (!isInstalled) {
+      return null;
```

Comment:
Should this fail more loudly if the reason we can't find devices is that the tool isn't installed like we expect?

Reply:
I've seen GitHub issues where people are trying to run `-d chrome` or for a particular Android device, and they are getting crashes because their personal phone is plugged in but isn't set up for development.  I don't think we should make people have Xcode installed if they aren't doing iOS development, and we can't really tell what their intentions are at the point we're trying to find a particular device identifier.

Reply:
Ah, okay, yeah that makes sense.